File: C/heap.c
#include <heap.h>

/** TODO: implement this **/


static void* heap_start;

/**
 *
 * Sets the start address of the heap
 * @param start the start address
 *
**/
void Sharpen_Heap_Init_1(void* start)
{
	heap_start = start;
}

/**
 *
 * Allocates a piece of memory on the heap
 * @param  size the size
 * @return the pointer
 *
**/
void* Sharpen_Heap_Alloc_1(int32_t size)
{
	void* ret = heap_start;
	heap_start = (void*) ((int) heap_start + size);
	return ret;
}

/**
 *
 * Allocates a piece of aligned memory on the heap
 * @param  alignment the alignment
 * @param  size the size
 * @return the pointer
 *
**/
void* Sharpen_Heap_AlignedAlloc_2(int32_t alignment, int32_t size)
{
	(void) alignment;
	return Sharpen_Heap_Alloc_1(size);
}

/**
 *
 * Frees a piece of code
 * @param ptr the pointer
 *
**/
void Sharpen_Heap_Free_1(void* ptr)
{
	(void) ptr;
}

File: user/shell/Shell/Sharpen/Memory/Heap.cs
namespace Sharpen.Memory
{
    unsafe class Heap
    {
        /// <summary>
        /// Allocates a block of memory
        /// </summary>
        /// <param name="size">The size of the memory block</param>
        /// <returns>The pointer to the allocated memory</returns>
        public static extern void* Alloc(int size);

        /// <summary>
        /// Frees a block of memory
        /// </summary>
        /// <param name="ptr">The pointer of the memory block</param>
        public static extern void Free(void* ptr);
    }
}
